core/src/main/java/hudson/model/AllView.java

```diff
@@ -89,13 +116,70 @@ protected void submit(StaplerRequest req) throws IOException, ServletException,
         // noop
     }
 
+    /**
+     * Corrects the name of the {@link AllView} if and only if the {@link AllView} is the primary view and
+     * its name is one of the localized forms of {@link Messages#_Hudson_ViewName()} and the user has opted in to
+     * fixing the view name by setting the system property {@code hudson.mode.AllView.JENKINS-38606} to {@code true}.
+     * Use this method to round-trip the primary view name, e.g.
+     * {@code primaryView = applyJenkins38606Fixup(views, primaryView)}
+     * NOTE: we can only fix the localized name of an {@link AllView} if it is the primary view as otherwise urls
+     * would change, whereas the primary view is special and does not normally get accessed by the
+     * {@code /view/_name_} url.
```

Comment:
Yes, actually the new item link has switched from `/view/All/newJob` to `/view/all/newJob` (note the lowercase in the default view name).